client/src/model/data_layer.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Class representing a data layer on the map.
```

Comment:
Optional nit: Let's not do "Class represents"... for comments since it seems repetitive. I would instead say what a data layer, conceptually, is

Reply:
Done.